Source/scrollrt.cpp

```diff
@@ -692,32 +691,29 @@ static void scrollrt_draw(int x, int y, int sx, int sy, int chunks, int dPieceRo
 				dst = &gpBuffer[sx + sy * BUFFER_WIDTH];
 				pMap = &dpiece_defs_map_2[x][y];
 				cel_transparency_active = (BYTE)(nTransTable[level_piece_id] & TransList[dTransVal[x][y]]);

```

Comment:
To fix the transparency issues with town, you can just do:
`if ( leveltype == DTYPE_TOWN) cel_transparency_active = FALSE;`

Reply:
@galaxyhaxz i already dealt with it. The issue was that the globals weren't cleared when going to town, simply calling DRLG_InitTrans mans that there isn't any need for a special case https://github.com/diasurgical/devilutionX/pull/361/commits/2a347455b64a6cd39c0bb841d6add2161a63e94f

Reply:
I get that but making sure the bool is false will always make sure the renderer never tries to draw transparent tiles in town. Of course if they were added in the future, it would make sense.

Reply:
Yeah I'm aiming at having it be as feature rich possible and as few special cases as possible

Reply:
Ah gotcha, "feature rich" is a phrase not spoken in vanilla. It's a naughty word there :P

Reply:
Hopfully we can have a level that transitions from cathedral to hell at some point :)